Mention mthree on configure-error-mitigation

abbycross opened the pull request:
Closes #2104

abbycross commented on docs/guides/configure-error-mitigation.mdx:
```diff
@@ -58,7 +58,9 @@ applied at each resilience level.
-You can use resilience levels to specify error mitigation techniques, or you can set custom techniques individually as described in [Custom error settings.](#advanced-error) You cannot specify resilience levels in Sampler. However, you can set custom techniques individually.
+You can use resilience levels to specify error mitigation techniques, or you can set custom techniques individually as described in [Custom error settings.](#advanced-error) 
+
+Sampler does not support error mitigation, but you can use the [mthree](https://qiskit.github.io/qiskit-addon-mthree/) (matrix-free measurement mitigation) package to do error mitigation locally.
```
linking to the ecosystem docs here - could link to the github repo instead or in addition - lmk if there is a preference for one of those options.

jyu00 replied:
I find it a bit odd to see this paragraph under `Configure Estimator with resilience levels`, since this is not related to resilience levels. However, I cannot decide if it's better under `Custom error settings, since the code example for Sampler there shows error _suppression_, not mitigation.

beckykd replied:
I can't figure out how to add a comment to a part of the file that wasn't edited, but maybe this would work: 
1 - At the top where we say "Primitives support several error mitigation techniques...", change that to "Estimator supports..." (since sampler doesn't support any)
2 - Move this m3 content after that updated paragraph as a note.

abbycross replied:
I was going to suggest it being a note too. @jyu00 do you have a preference for one of Becky's suggestions?

jyu00 replied:
I think Becky was suggesting we do both? Change the top to say just Estimator, and follow that with a note saying you can use M3 for Sampler. I do like this approach.

abbycross replied:
ah yes, I don't know why I assumed it was one or the other 🤣 both it is!

Eric-Arellano commented on docs/guides/configure-error-mitigation.mdx:
```diff
@@ -58,7 +58,9 @@ applied at each resilience level.
-You can use resilience levels to specify error mitigation techniques, or you can set custom techniques individually as described in [Custom error settings.](#advanced-error) You cannot specify resilience levels in Sampler. However, you can set custom techniques individually.
+You can use resilience levels to specify error mitigation techniques, or you can set custom techniques individually as described in [Custom error settings.](#advanced-error) 
+
+Sampler does not support error mitigation, but you can use the [mthree](https://qiskit.github.io/qiskit-addon-mthree/) (matrix-free measurement mitigation) package to do error mitigation locally.
```
Should it be "The Sampler primitive"?

abbycross replied:
We use them interchangeably

beckykd commented on docs/guides/configure-error-mitigation.mdx:
```diff
@@ -58,7 +58,9 @@ applied at each resilience level.
-You can use resilience levels to specify error mitigation techniques, or you can set custom techniques individually as described in [Custom error settings.](#advanced-error) You cannot specify resilience levels in Sampler. However, you can set custom techniques individually.
+You can use resilience levels to specify error mitigation techniques, or you can set custom techniques individually as described in [Custom error settings.](#advanced-error) 
+
+Sampler does not support error mitigation, but you can use the [mthree](https://qiskit.github.io/qiskit-addon-mthree/) (matrix-free measurement mitigation) package to do error mitigation locally.
```
```suggestion
Sampler does not support error mitigation, but you can use the [mthree](https://qiskit.github.io/qiskit-addon-mthree/) (matrix-free measurement mitigation) package to perform error mitigation locally.
```

beckykd replied:
I'm not a giant fan of "perform" either, but "do" sounds weird to me.

abbycross replied:
I definitely like `perform` better than `do`. 👍

beckykd reviewed (approved):
Just a couple of suggestions.  Otherwise, LGTM

abbycross commented:
@jyu00 ready for final review... I removed the text from the bottom and put it in a note near the top -- also changed primitives to estimator:
<img width="865" alt="Screenshot 2024-10-30 at 2 58 39 PM" src="https://github.com/user-attachments/assets/b9f89f24-523e-45ee-a777-0f9d06e37388">
